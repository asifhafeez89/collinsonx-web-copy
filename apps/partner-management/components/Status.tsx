import { Box, Text } from '@collinsonx/design-system/core';
import {
  Confirmed,
  Pending,
  Declined,
} from '@collinsonx/design-system/assets/icons';

export type type = 'warning' | 'success' | 'danger';
export type typeProps = {
  type: type;
  children: JSX.Element | string;
};

const styles = {
  padding: '8px 16px',
  borderRadius: 4,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  gap: 10,
  height: 44,
  fontSize: 14,
  fontWeight: 500,
  color: '#000000',
};

interface Props {
  children: JSX.Element | string;
}

const statusMap = {
  warning: ({ children }: Props) => (
    <Box
      sx={{
        ...styles,
        background: '#FEF3DA',
        border: '2px solid #FAB005',
      }}
    >
      <Pending />
      <Text>{children}</Text>
    </Box>
  ),
  success: ({ children }: Props) => (
    <Box
      sx={{
        ...styles,
        background: '#EEF9E7',
        border: '2px solid #54C50D',
      }}
    >
      <Confirmed />
      <Text>{children}</Text>
    </Box>
  ),
  danger: ({ children }: Props) => (
    <Box
      sx={{
        ...styles,
        background: '#FDECEC',
        border: '2px solid #F03E3E',
      }}
    >
      <Declined />
      <Text>{children}</Text>
    </Box>
  ),
};

export default function type({ type, children }: typeProps) {
  const Component = statusMap[type];
  return <Component>{children}</Component>;
}
