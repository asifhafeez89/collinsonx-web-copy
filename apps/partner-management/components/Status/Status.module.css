.base {
  padding: 8px 16px;
  border-radius: 4;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 10;
  height: 44;
  font-size: 14;
  font-weight: 500;
  color: #000000;
}
.statusPending {
  background: #fef3da;
  border: 2px solid #fab005;
}
.statusConfirmed {
  background: #eef9e7;
  border: 2px solid #54c50d;
}
.statusDeclined {
  background: #fdecec;
  border: 2px solid #f03e3e;
}
