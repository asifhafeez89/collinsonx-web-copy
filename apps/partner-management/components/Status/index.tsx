import { Box, Text } from '@collinsonx/design-system/core';
import cx from 'clsx';
import {
  Confirmed,
  Pending,
  Declined,
} from '@collinsonx/design-system/assets/icons';

import classes from './Status.module.css';

export type type = 'warning' | 'success' | 'danger';
export type typeProps = {
  type: type;
  children: JSX.Element | string;
};

interface Props {
  children: JSX.Element | string;
}

const statusMap = {
  warning: ({ children }: Props) => (
    <Box className={cx([classes.base, classes.statusPending])}>
      <Pending />
      <Text>{children}</Text>
    </Box>
  ),
  success: ({ children }: Props) => (
    <Box className={cx([classes.base, classes.statusConfirmed])}>
      <Confirmed />
      <Text>{children}</Text>
    </Box>
  ),
  danger: ({ children }: Props) => (
    <Box className={cx([classes.base, classes.statusDeclined])}>
      <Declined />
      <Text>{children}</Text>
    </Box>
  ),
};

export default function type({ type, children }: typeProps) {
  const Component = statusMap[type];
  return <Component>{children}</Component>;
}
