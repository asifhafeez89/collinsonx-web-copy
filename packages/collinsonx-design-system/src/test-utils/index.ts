import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
