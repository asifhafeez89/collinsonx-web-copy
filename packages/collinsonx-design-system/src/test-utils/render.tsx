import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
