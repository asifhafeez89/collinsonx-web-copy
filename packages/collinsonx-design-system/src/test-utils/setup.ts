import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
