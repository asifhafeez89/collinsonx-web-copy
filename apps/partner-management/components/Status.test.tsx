import React from 'react';
import { render } from '@collinsonx/design-system/test-utils';
import Status from './Status';
jest.mock('@collinsonx/design-system/assets/icons', () => {
  return {
    __esModule: true,
    Pending: () => [],
    Confirmed: () => [],
    Declined: () => [],
  };
});
test('renders Type component with warning type', () => {
  const component = render(<Status type={'warning'}>warning</Status>);
  expect(component).toMatchSnapshot();
});
test('renders Type component with success type', () => {
  const component = render(<Status type={'success'}>success</Status>);
  expect(component).toMatchSnapshot();
});
test('renders Type component with danger type', () => {
  const component = render(<Status type={'danger'}>danger</Status>);
  expect(component).toMatchSnapshot();
});
