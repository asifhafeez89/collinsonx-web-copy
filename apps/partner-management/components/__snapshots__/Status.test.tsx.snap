// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Type component with danger type 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  padding: 8px 16px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 10px;
  height: 44px;
  font-size: 14px;
  font-weight: 500;
  color: #000000;
  background: #FDECEC;
  border: 2px solid #F03E3E;
}

.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1 emotion-2"
        >
          danger
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  padding: 8px 16px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 10px;
  height: 44px;
  font-size: 14px;
  font-weight: 500;
  color: #000000;
  background: #FDECEC;
  border: 2px solid #F03E3E;
}

.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1 emotion-2"
      >
        danger
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Type component with success type 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  padding: 8px 16px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 10px;
  height: 44px;
  font-size: 14px;
  font-weight: 500;
  color: #000000;
  background: #EEF9E7;
  border: 2px solid #54C50D;
}

.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1 emotion-2"
        >
          success
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  padding: 8px 16px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 10px;
  height: 44px;
  font-size: 14px;
  font-weight: 500;
  color: #000000;
  background: #EEF9E7;
  border: 2px solid #54C50D;
}

.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1 emotion-2"
      >
        success
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Type component with warning type 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  padding: 8px 16px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 10px;
  height: 44px;
  font-size: 14px;
  font-weight: 500;
  color: #000000;
  background: #FEF3DA;
  border: 2px solid #FAB005;
}

.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1 emotion-2"
        >
          warning
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  padding: 8px 16px;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 10px;
  height: 44px;
  font-size: 14px;
  font-weight: 500;
  color: #000000;
  background: #FEF3DA;
  border: 2px solid #FAB005;
}

.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1 emotion-2"
      >
        warning
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
